src/Builder/CMakeLists.txt

```diff
@@ -8,14 +8,19 @@ target_include_directories(OMBuilder PRIVATE ${ONNX_MLIR_SRC_ROOT})
 target_include_directories(OMBuilder PRIVATE ${CMAKE_BINARY_DIR})
 target_include_directories(OMBuilder PRIVATE ${ONNX_MLIR_BIN_ROOT})
 
+# This will cause onnx to be built. More importantly, some variable definitions
+# when building onnx such as -DONNX_ML=1 -DONNX_NAMESPACE=onnx will be carried over
+# when compiling FrontendDialectHelper.cpp, etc.
 target_link_libraries(OMBuilder
-        OMONNXOps
         onnx
-        ${MLIRLibs}
-        curses
         mpark_variant)
 target_include_directories(OMBuilder
                            PUBLIC
                            ${ONNX_MLIR_SRC_ROOT}/third_party/onnx
                            ${ONNX_MLIR_SRC_ROOT}/third_party/variant
                            ${ONNX_MLIR_SRC_ROOT})
+# If you add onnx here, it will also cause onnx to be built. However, some
+# variable definitions when building onnx such as -DONNX_ML=1 -DONNX_NAMESPACE=onnx
+# will NOT be carried over when compiling FrontendDialectHelper.cpp, etc. so
+# the compilation will fail.
+add_dependencies(OMBuilder OMONNXOps OMShapeInferenceOpInterface)
```

Comment:
I do not quite understand the changes done to this file. Could you explain a bit more?

Reply:
Compilation for `FrontendDialectHelper.cpp` fails because it ultimately includes `third_party/onnx/onnx/onnx_pb.h`, which has this code:
```
#ifdef ONNX_ML
#include "onnx/onnx-ml.pb.h"
#else
#include "onnx/onnx.pb.h"
#endif
```
When `onnx` is specified in `target_link_libraries`, somehow cmake generates `-DONNX_ML=1` so `onnx-ml.pb.h` gets included and the compilation works. But if you specify `onnx` in `add_dependencies`, somehow cmake does not generate `-DONNX_ML=1` so `onnx.pb.h` gets include and the compilation fails because `onnx.pb.h` does not exist. I was trying to remove the `target_link_libraries` and move `onnx` and `mpark_variant` into `add_dependencies` but couldn't do it because of this. It looks like dependency targets specified in `target_link_libraries` and `add_dependencies` get compiled somewhat differently. So I added the comments to record this.